### DISCUSSION TITLE
Issue with Vercel serverless functions in Angular app

### DISCUSSION THREAD

**Discussion by MikeJWms** [vercel/community#1972] [Help] [answered]
### Summary

Hi,

I'm developing an Angular app and using Vercel's serverless functions. I followed the [Serverless quickstart guide](https://vercel.com/docs/concepts/functions/serverless-functions/quickstart) and placed the /api folder in my projects ```/src``` directory, along with a ```/server``` folder for helper functions. When running locally with ```vercel dev```, everything works as expected.

However, when trying to deploy the angular app would build, but the serverless functions would not. 

I tried changing the root directory in Vercel project settings from '' to 'src', but it resulted in an error: 
```
sh: ng: command not found
Error: Command "ng build" exited with 127
```

So, I reverted this change.

I then tried moved everything from the src directory to the top level project directory, which solved the issue of serverless functions not building. However, now I'm getting errors that the serverless environment can't find the imported functions exported from the ```/server``` folder.

I prefer not to have all my src files in the project root. Can anyone provide a working configuration for the following structure?

/angular-project-root
      |--> /src
                  |--> /api (vercel serverless functions)
                  |--> /app (angular app)
                  |--> /server (helper functions for the serverless functions to use)

Any help would be appreciated. Thank you.

**UPDATE:** After making a minimal reproduction repo I've realized that I missed one detail in the quickstart docs - ```/api``` should be created in the project root, not the ```/src``` folder. I am now having success with both angular and serverless functions deploying.

The problem I'm facing now is with importing helper functions into my serverless functions. For example, when calling the endpoint in the example repo, the following error occurs: 
```
Error: Cannot find module '/var/task/server/helpers' imported from /var/task/api/handler.js
```

### Example

Here is the example repo: https://github.com/MikeJWms/angular-vercel

**Comment by TooTallNate** [comment]
Could you share a minimal reproduction of what you're trying to achieve?

**Reply by MikeJWms** [reply]
Hi @TooTallNate, I've added an update to the original post. Here is an example repo: https://github.com/MikeJWms/angular-vercel

**Reply by TooTallNate** [reply, marked as answer]
Thanks for that! I see what's happening now. This is due to the `"type": "module"` in the `package.json` file, which enables ESM import mode in Node.js. In this mode, you need to include the file extension when importing files. Note that even though it's a TypeScript file, you use `.js` file extension, since that's what TypeScript ends up emitting.

I've created a pull request which is working for me: https://github.com/MikeJWms/angular-vercel/pull/1

**Reply by MikeJWms** [reply]
Thanks for your help @TooTallNate! Sometimes it's the littlest things that cause the biggest problems.
